### ISSUE TITLE
Modal - heading level prop

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-react#1736]
Currently, modals in core use `<h1>` as the heading level, but the react components have this set to `<h3>`.

There is some discussion related to what the recommended levels should be here:
https://github.com/w3c/aria-practices/issues/551#issuecomment-365134527

For this issue, we should provide a prop to allow consumers to change the level, and use a default value based on the recommendation by the aria-practices group.

For our default I'd suggest `<h1>` unless someone has a strong reason for `<h2>`.
